openwisp_monitoring/device/admin.py

```diff
@@ -63,10 +63,32 @@ class CheckInline(GenericStackedInline):
     fields = ['check_type', 'is_active']
     readonly_fields = ['check_type']
 
+    def _set_check_inline(self):
+        self.fields = []
+        self.readonly_fields = ['created', 'modified']
```

Comment:
what is this doing? I don't understand.

Reply:
`CheckInline` (with current fields ['check_type', 'is_active'])

![Screenshot from 2022-08-14 15-36-00](https://user-images.githubusercontent.com/56113566/184532079-14067716-aaf4-4cf8-8eb9-2a58b415de90.png)

`CheckInline` (with all fields [])

I think this will be more useful for the users because here they have more options to configure e.g. `checks params, name` etc & also it's similar to general check page, Although we can remove `description` field if required.
 
![Screenshot from 2022-08-14 15-39-41](https://user-images.githubusercontent.com/56113566/184532191-e4f9511e-7d02-4900-9bd4-3bd5b23c28fd.png)

Reply:
why do you set it this way instead of just setting it at class level? I can't make sense of it.

Reply:
I do this way because we only want to show these fields when the user has `check_inline` permissions if we set those at the class level then those form fields are also visible for `check model` permissions that have already access to `check general page`.